packages/scheduler/src/forks/SchedulerHostConfig.default.js

```diff
@@ -83,8 +83,12 @@ if (
   const Date = window.Date;
   const setTimeout = window.setTimeout;
   const clearTimeout = window.clearTimeout;
-  const requestAnimationFrame = window.requestAnimationFrame;
-  const cancelAnimationFrame = window.cancelAnimationFrame;
+  // Explicitly bind local references to `window`, in order to support
+  // environments where the global `this` is not the window (e.g extensions)
+  const requestAnimationFrame =
+    window.requestAnimationFrame && window.requestAnimationFrame.bind(window);
```

Comment:
Do you mind making these more explicit, i.e.

```
const requestAnimationFrame =
    typeof window.requestAnimationFrame === 'function' ? window.requestAnimationFrame.bind(window) : undefined;
```

I know it's super unlikely but would be nice to recover gracefully on startup if globals are clobbered with invalid values.

Reply:
Good point, done.